ethereum/core/src/test/java/org/hyperledger/besu/ethereum/transaction/TransactionSimulatorTest.java

```diff
@@ -923,6 +926,47 @@ public void shouldSimulateLegacyTransactionWhenBaseFeeNotZero() {
     assertThat(result.get().isSuccessful()).isTrue();
   }
 
+  @Test
+  public void shouldGuessDelegateCodeTransactionTypeWhenAuthorizationsPresent() {
+    final CodeDelegation delegation =
+        new CodeDelegation(BigInteger.ONE, Address.fromHexString("0x1"), 42L, FAKE_SIGNATURE);
+
+    final CallParameter callParameter =
+        legacyTransactionCallParameterBuilder()
+            .codeDelegationAuthorizations(List.of(delegation))
+            .build();
```

Comment:
Can we move this to a `codedelegatoinTransactionCallParameterBuilder`? we have a builder for all the other tx types

Reply:
Done.